src/main/com/intellij/lang/jsgraphql/ide/project/graphqlconfig/GraphQLConfigManager.java

```diff
@@ -425,6 +428,18 @@ public JSGraphQLEndpointSchemaConfiguration getEndpointLanguageConfiguration(@No
         }
     }
 
+    @Nullable
+    public GraphQLConfigSecurity getSSLConfiguration() {
+        try {
+            readLock.lock();
+            Map<VirtualFile, GraphQLConfigData> configEntries = getService(myProject).getConfigurationsByPath();
+            GraphQLConfigSecurity sslConfig = new ObjectMapper().convertValue(configEntries.get(myProject.getBaseDir()).extensions.get(SSL_EXTENSION), GraphQLConfigSecurity.class);
```

Comment:
Oh, I'm not sure if it's correct. First,`com.intellij.openapi.project.Project#getBaseDir` is deprecated, it has an alternative `com.intellij.openapi.project.ProjectUtil#guessProjectDir`. But what's more important is that it looks like we need to use a corresponding config for request configuration, not just a random one from the project directory (`com.intellij.lang.jsgraphql.ide.project.graphqlconfig.GraphQLConfigManager#getEndpoints` as a reference).

There are two usages of `createHttpClient` now. For example, here https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/blob/8f41775c4f71c97b52cc9b826277ee2828823c4d/src/main/com/intellij/lang/jsgraphql/ide/editor/GraphQLIntrospectionService.java#L484 you can get a configuration file using something like `GraphQLConfigManager.getService(myProject).getConfigurationsByPath().get(introspectionSourceFile.getParent())`. `getParent` call is needed because configs are cached by the containing directory, not the file itself. So, here we could call something like `prepareRequestContext(@Nullable GraphQLConfigData configData)` and then pass it to `createHttpClient` method.

What do you think?

Reply:
Updated as suggested. Added a new method to parse the security config. The createHttpClient now takes a nullable security config which is created without using an object mapper. Also updated both uses of createhttpClient. Let me know if that looks reasonable. Thanks for all the comments and feedback so far.